gluten-core/src/main/scala/io/glutenproject/extension/columnar/TransformHintRule.scala

```diff
@@ -62,16 +61,25 @@ object TransformHints {
 
   val DEBUG = false
 
-  def isAlreadyTagged(plan: SparkPlan): Boolean = {
-    plan.getTagValue(TAG).isDefined
+  def isNotTransformable(plan: SparkPlan): Boolean = {
+    getHintOption(plan) match {
+      case Some(TRANSFORM_UNSUPPORTED(_, _)) => true
+      case _ => false
+    }
   }
 
+  /**
+   * NOTE: To be deprecated. Do not create new usages of this method.
+   *
+   * Since it's usually not safe to consider a plan "transformable" during validation phase. Another
+   * validation rule could turn "transformable" to "non-transformable" before implementing the plan
+   * within Gluten transformers.
+   */
   def isTransformable(plan: SparkPlan): Boolean = {
-    plan.getTagValue(TAG).get.isInstanceOf[TRANSFORM_SUPPORTED]
-  }
-
-  def isNotTransformable(plan: SparkPlan): Boolean = {
-    plan.getTagValue(TAG).get.isInstanceOf[TRANSFORM_UNSUPPORTED]
+    getHintOption(plan) match {
+      case None => true
+      case _ => false
+    }
```

Comment:
If `isNotTransformable(plan)` returns true, it means plan cannot be offloaded for sure. If `isNotTransformable(plan)` returns false, it means the plan could or could not be offloaded. Is this correct?

Reply:
> If `isNotTransformable(plan)` returns true, it means plan cannot be offloaded for sure. If `isNotTransformable(plan)` returns false, it means the plan could or could not be offloaded. Is this correct?

That's correct.

Reply:
Could we add some comments for this function to clarify? Thanks.

Reply:
Sure, will add some in next commit. Thanks.